src/repr/src/row/encode.rs

```diff
@@ -1364,7 +1364,12 @@ impl ColumnEncoder<Row> for RowColumnarEncoder {
             .iter()
             .zip_eq(encoders)
             .map(|((col_idx, _col_name), encoder)| {
-                let nullable = encoder.nullable;
+                // Note: We mark all columns as nullable at the Arrow/Parquet level because it has
+                // a negligible performance difference, but it protects us from unintended
+                // nullability changes in the columns of SQL objects.
+                //
+                // See: <https://github.com/MaterializeInc/database-issues/issues/2488>
+                let nullable = true;
```

Comment:
We need to make the same change in `DatumColumnEncoder` I believe for nested structs!

Reply:
Great catch! Made this change, will refactor away `DatumEncoder` in a follow up